### PR TITLE
Added a new debug host field to AWS plugin global settings

### PR DESCRIPTION
**Pull request by tbtmuse**
## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Description
Added a new text field above cloud debug called 'Local debug host', this value is saved on a global AWS plugin level and is then used by each debug extension in place of the hardcoded "localhost"  host. 

In the event that the local debug host field is left blank, the debug host defaults to "localhost". 

## Motivation and Context
The hard coded debugger host in the debug extensions support files, made it impossible to debug lamda functions locally when the debug host is not accessible via "localhost". This change exposes that setting allowing better flexibility. 

## Related Issue(s)
#642 #784 

## Testing
Tested changes by running local gradle tests and making sure those passed, and by building the plugin and running it in intellij ultimate using the debugging locally guide. Also testing both scenarios where no value is specified and where value is specified.

Testing environment:
Windows 10 - Slow ring build 19033
SAM CLI: 0.31.1
AWS CLI: 1.16.280
Pycharm & Intellij Ultimate

## Screenshots
screenshot

## Checklist
- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `gradlew check` succeeds
- [x] My code follows the code style of this project
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] A short description of the change has been added to the **CHANGELOG**

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/SamDebugSupport.kt [hunk 1]
```diff
@@ -26,17 +27,29 @@ interface SamDebugSupport {
...
         state: SamRunningState,
+        debugHost: String,
         debugPorts: List<Int>
     ): XDebugProcessStarter?
 
     fun isSupported(): Boolean = true
 
+    fun getDebugHost(): String {
+
+        val localDebugHost = LocalDebugSettings.getInstance().localDebugHost
```
Comment (kiiadi) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/SamDebugSupport.kt]: Could do this in a single statement with the Elvis operator.

Comment (tbtmuse) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/SamDebugSupport.kt]: Noted, I can convert this to make use of the Elvis operator here, I chose to KISS, however will update the PR.

**Comment by tbtmuse** [comment]
@kiiadi thanks for taking the time to review.

To my understanding the Sam executable setting is unlikely to change per run/debug configuration and so is the location of the cloud debug cli executable, and this is why they are global (I could be very wrong). 

I also considered the debugger host to be a setting that is unlikely to change per run/debug configuration, this so why I made it global.

I don't see this working with the ECS cloud debugger feature as they both seem very specify to the type of code, lambda VS code sitting in an EC2 ibsta

**Comment by kiiadi** [comment]
Just circling back here - I want to understand a bit more about your setup to see why we need this? Is there a way we can 'infer' the "local" connection? e.g would replacing hard-coded `localhost` with `InetAddress.getLocalHost()` and/or `InetAddress.getLoopbackAddress()` work instead?

**Comment by tbtmuse** [comment]
The only way I can see this working without allowing a user to explicitly set the IP of their debug host, is to determine if Intellij is connected to Docker via Docker Machine and the grabbing the IP of the Docker host somehow.

InetAddress.getLocalHost() and/or InetAddress.getLoopbackAddress() will not be good enough.

**Comment by abrooksv** [comment]
Are you able to connect to Docker from the IDE?

**Comment by tbtmuse** [comment]
Yes indeed, I am able to connect from the IDE.

**Comment by abrooksv** [comment]
Querying the Docker plugin isnt too difficult, the `com.intellij.docker.DockerCloudConfiguration` contains the API URL, so we should be able to get that and then say `.host()` on the URI if the URI is not a unix socket.

**Comment by tbtmuse** [comment]
Yes it does contain the same IP I need to debug.

**Comment by abrooksv** [comment]
#1484 stores it in the run config and queries it from Docker plugin

**Comment by abrooksv** [comment]
@tbtmuse Thank you for the pull request and bringing this to our attention. I have merged #1484 that bases this on your docker settings, you can select the host on the SAM CLI page of the Run Configuration.

**Pull request closed without merging**